### ISSUE TITLE
Error when trying to download a release in a GitHub action

### ISSUE THREAD

**Issue by boosty** [heroku/heroku-jvm-application-deployer#329]
Thanks for providing this tool! 

I am trying to download the released jar from this repo in a GitHub action, using the GitHub CLI / `gh` command:

```bash
gh release download \
	-R "heroku/heroku-jvm-application-deployer" \
	-p "*.jar" \
	-O ".downloads/heroku-jvm-application-deployer.jar" \
	"v4.0.6"
```

The command works on my local machine, but not when running in a GitHub action:

> HTTP 403: Although you appear to have the correct authorization credentials, the `heroku` organization has an IP allow list enabled, and your IP address is not permitted to access this resource. (https://api.github.com/repos/heroku/heroku-jvm-application-deployer/releases/tags/v4.0.6)

Is the IP allow list on your side intentional?

Thanks for your help!

**Comment by Malax** [comment]
Hi @boosty!

There is IP allow listing enabled, but I don't think its supposed to affect read-only access. Especially since it's totally fine to download the release from the GitHub website. This might be a bug on GitHubs side or an unintentional side-effect of some API being affected by the IP allow listing that is used by `gh` to fetch the release.

I'll try to reproduce the issue and will update this issue with my findings. Thanks for bringing this to my attention!

**Comment by Malax** [comment]
I just ran the command from your initial message from a computer with an external IP address with a GitHub account that is not part of the `heroku` organization. It completed successfully with the most recent `gh` release.

Can you try again (with the latest `gh` version if you're not using it already)? Maybe this was just a hiccup on GitHub's side.

**Comment by boosty** [comment]
Hey @Malax, thanks for the quick feedback!

The command works on two different machines we tested it with (different GH user accounts), but not as part of a GitHub action run.

Did you try running the command from within a GitHub action, on a GitHub-hosted runner?

**Comment by Malax** [comment]
Ah, I see. Sorry, I did miss fact you're talking about a GitHub action run. I've yet to try that.

**Comment by Malax** [comment]
Hi @boosty!

I finally had the time to test this on GHA. I created a minimal expample with the expectation I'd need to contact GitHub support and to ask them about it. However, it just worked on both my `malax` account which has write access to this repository as well as with new GitHub account without any special permissions.

The test run can be seen here: https://github.com/malax-noheroku/reproduce-gh-release-issue/actions/runs/9612052061/job/26511923229 and this is the workflow file for that run: https://github.com/malax-noheroku/reproduce-gh-release-issue/actions/runs/9612052061/workflow

Do you see anything in there that is different from your workflow? In the end, this problem seems to be specific to your workflow and/or account setup. If you cannot figure this out I'd suggest asking GitHub about it. In the meantime, hardcoding the download URL seems to be an okay, albeit less elegant, workaround to unblock you for now.

Please let me know if you're able to solve this issue and happy to assist further!

**Comment by boosty** [comment]
@Malax Thanks for all your help!

I ran your example from https://github.com/malax-noheroku/reproduce-gh-release-issue/actions/runs/9612052061/workflow in our repo, but it results in the same 403 error: 

> HTTP 403: Although you appear to have the correct authorization credentials, the `heroku` organization has an IP allow list enabled, and your IP address is not permitted to access this resource

I went for a curl-based download using a hardcoded URL for now. This works and is ok for us for now. I might contact GitHub though, would be interesting to understand where the problem is.

Thanks again!
